Dockerfile.test

```diff
@@ -0,0 +1,27 @@
+# Take in a runtime image to use for the base system
+# Expects an alpine image
+ARG RUNTIME_IMAGE
+
+# Use the docker image provided via build arg
+FROM $RUNTIME_IMAGE
+
+# Install the libraries we need for go and dockerize
+RUN apk add --no-cache curl git
+
+# Copy in the dockerize utility
+ARG DOCKERIZE_VERSION=0.6.0
+RUN curl -sL https://github.com/jwilder/dockerize/releases/download/v$DOCKERIZE_VERSION/dockerize-alpine-linux-amd64-v$DOCKERIZE_VERSION.tar.gz | tar -xzC /usr/local/bin
+
+# Copy project into the image
+COPY . /src/github.com/fauna/faunadb-go
+
+# Shift over to the project, set GOPATH, and fetch dependencies
+WORKDIR /src/github.com/fauna/faunadb-go/faunadb
+ENV GOPATH=/
+RUN go get -t -v
+
+# Define the default variables for the tests
+ENV FAUNA_ROOT_KEY=secret FAUNA_ENDPOINT=https://db.fauna.com
+
+# Run the tests (after target database is up)
+ENTRYPOINT dockerize -wait "$FAUNA_ENDPOINT/ping" -timeout 30s go test -v
```

Comment:
Actually, I have a question. Why did you decide to check fauna's state in the dockerfile itself rather than in the docker-compose you have in Jenkins?

Reply:
Is there a way to actually hold off on starting the tests until FaunaDB is up from docker compose? I didn't see one.

Reply:
I _think_ you can move it to the docker compose file and clean the docker image a little bit: https://docs.docker.com/compose/startup-order/

I haven't tested it, though.

Reply:
I didn't want to have to duplicate the test command from the dockerfile into the compose file, but if that's not too much of a concern, I can move the dockerize + test command into the compose file and just have the test command here.

Reply:
I realized now that even if we attempt to use dockerize with docker compose we'd need the dockerize script installed in the image itself so, moving it out doesn't buy us anything.

Reply:
Yeah... :( Should I just leave this all in place?